xbmc/filesystem/AddonsDirectory.cpp

```diff
@@ -788,11 +788,29 @@ void CAddonsDirectory::GenerateAddonListing(const CURL &path,
     bool disabled = CServiceBroker::GetAddonMgr().IsAddonDisabled(addon->ID());
     bool hasUpdate = outdated.find(addon->ID()) != outdated.end();
     bool fromOfficialRepo = CServiceBroker::GetAddonMgr().IsFromOfficialRepo(addon);
+    AddonOriginType addonOriginType = CServiceBroker::GetAddonMgr().GetAddonOriginType(addon);
 
     pItem->SetProperty("Addon.IsInstalled", installed);
     pItem->SetProperty("Addon.IsEnabled", installed && !disabled);
     pItem->SetProperty("Addon.HasUpdate", hasUpdate);
     pItem->SetProperty("Addon.IsFromOfficialRepo", fromOfficialRepo);
+    pItem->SetProperty("Addon.IsBinary", addon->IsBinary());
+
+    std::string addonOriginTypeProperty;
+    switch (addonOriginType)
+    {
+      case AddonOriginType::SYSTEM:
+        addonOriginTypeProperty = "system";
+        break;
+      case AddonOriginType::REPOSITORY:
+        addonOriginTypeProperty = "repository";
+        break;
+      case AddonOriginType::MANUAL:
+      default:
+        addonOriginTypeProperty = "manual";
+        break;
```

Comment:
please do not use hardcoded strings. these need to be localized.

Reply:
The idea here was to have them enumerated instead of booleans. It should allow logic in the skin such as if Add.OriginType == manual, show this icon. Have I used the wrong approach?

Reply:
yes, since this is an infolabel (i.e. it can be displayed as a text label in the skin), those strings need to be added to the language file.
skins can then still use the Addon.OriginType == $LOCALIZE[12345] logic to display an icon.

Reply:
Ok, coo, thanks!

Reply:
Do I document those localisation IDs somewhere?

Reply:
nope, looks good enough as is to me.